### PR TITLE
Fix index/constraint lookup when preparing pg_restore list files.

### LINKED ISSUE #759: [SQLite 100] another row available
Migration fails to filter already processed constraints, resulting errors during  pg_restore of post-filtered.list

```
2024-04-24 14:06:24.496 42 SQLite catalog.c:7248            [SQLite]   select pid, start_time_epoch, done_time_epoch, duration     from summary    where indexoid = $1 or conoid = $2
2024-04-24 14:06:24.496 42 SQLite catalog.c:7627            [SQLite] 16978, 16978
2024-04-24 14:06:24.496 42 ERROR  catalog.c:7395            Failed to execute statement:   select pid, start_time_epoch, done_time_epoch, duration     from summary    where indexoid = $1 or conoid = $2
2024-04-24 14:06:24.496 42 ERROR  catalog.c:7396            [SQLite 100] another row available
2024-04-24 14:06:24.496 42 ERROR  catalog.c:7321            Failed to execute SQLite query, see above for details
2024-04-24 14:06:24.496 42 SQLite catalog.c:7248            [SQLite]   select oid, restore_list_name, kind     from filter    where oid = $1
2024-04-24 14:06:24.496 42 SQLite catalog.c:7627            [SQLite] 16978
2024-04-24 14:06:24.496 42 SQLite catalog.c:7248            [SQLite]   select oid, restore_list_name, kind     from filter    where restore_list_name = $1
2024-04-24 14:06:24.496 42 SQLite catalog.c:7627            [SQLite] public t_data_file t_data_file_pkey sa
```
The SQLite has got entries with same OIDs between indexoid and conoids 
```
sqlite> SELECT * FROM summary WHERE indexoid = 16978 OR conoid = 16978;
70||||16978|1713967577|1713967577|43||ALTER TABLE public.t_data_file ADD CONSTRAINT t_data_file_pkey PRIMARY KEY USING INDEX t_data_file_pkey
68|||16978||1713967583|1713967583|34||CREATE UNIQUE INDEX t_thumbnail_pkey ON testtenant01.t_thumbnail USING btree (id);
sqlite> 
sqlite> select * from s_index where oid=16978;
16978|testtenant01.t_thumbnail_pkey|testtenant01|t_thumbnail_pkey|testtenant01 t_thumbnail_pkey sa|16974|1|1|id|CREATE UNIQUE INDEX t_thumbnail_pkey ON testtenant01.t_thumbnail USING btree (id)
sqlite> select * from s_constraint where oid=16978;
16978|t_data_file_pkey|16449|0|0|PRIMARY KEY (id)
```

### PR DESCRIPTION
**Pull request by dimitri**
The Postgres catalog OIDs are unique per-catalog, not globally unique, and we could see the same OID being assigned to both an index and also a constraint.

To support the situation, we need to implement per-catalog lookup. In our case this is limited to indexes and constraints.

Fix #759

### CODE REVIEW

File: src/bin/pgcopydb/dump_restore.c [hunk 1]
```diff
@@ -31,23 +31,58 @@ static bool copydb_copy_database_properties_hook(void *ctx,
...
+			if (!summary_lookup_constraint(sourceDB, &indexSpecs))
+			{
+				/* errors have aleady been logged */
+				return false;
+			}
+
+			return indexSpecs.summary.doneTime > 0;
+		}
+
+		/* we don't have internal pgcopydb suppor for other objects */
```
Comment (hanefi) [inline src/bin/pgcopydb/dump_restore.c:77]: minor typo

```suggestion
		/* we don't have internal pgcopydb support for other objects */
```

File: src/bin/pgcopydb/dump_restore.c [hunk 2]
```diff
@@ -31,23 +31,58 @@ static bool copydb_copy_database_properties_hook(void *ctx,
...
+			SourceIndex index = { .indexOid = oid };
+			CopyIndexSpec indexSpecs = { .sourceIndex = &index };
+
+			if (!summary_lookup_index(sourceDB, &indexSpecs))
+			{
+				/* errors have aleady been logged */
+				return false;
+			}
+
+			return indexSpecs.summary.doneTime > 0;
```
Comment (hanefi) [inline src/bin/pgcopydb/dump_restore.c:60]: I like the fact that we changed our logic to decide whether or not an item is processed. It is definitely an improvement over whether or not the lookup exists in summary.

**Review by hanefi — APPROVED** [review]
1 minor typo. Otherwise good to merge.

**Comment by arajkumar** [comment]
@dimitri  I believe this fixes https://github.com/dimitri/pgcopydb/issues/772 as well.

**Comment by dimitri** [comment]
Thanks @arajkumar ; closed too.

**Pull request merged**